catalog/glue.go

```diff
@@ -122,33 +149,93 @@ func (c *GlueCatalog) LoadTable(ctx context.Context, identifier table.Identifier
 	return icebergTable, nil
 }
 
-func (c *GlueCatalog) CatalogType() CatalogType {
-	return Glue
-}
-
+// DropTable deletes an Iceberg table from the Glue catalog.
 func (c *GlueCatalog) DropTable(ctx context.Context, identifier table.Identifier) error {
-	return fmt.Errorf("%w: [Glue Catalog] drop table", iceberg.ErrNotImplemented)
+	database, tableName, err := identifierToGlueTable(identifier)
+	if err != nil {
+		return err
+	}
+
+	// Check if the table exists and is an Iceberg table.
+	_, err = c.getTable(ctx, database, tableName)
+	if err != nil {
+		return err
+	}
+
+	params := &glue.DeleteTableInput{
```

Comment:
does this not need a glue catalog id?

Reply:
a glue catalog id is used in pyiceberg and in the java implementation, so I would have assumed that we're using it for the go client too

Reply:
Added. Followed the Java pattern where we add catalog id in every request input. Event listener based approach from Python Iceberg might be ugly in Golang.